src/commands/farm.rs

```diff
@@ -42,17 +76,21 @@ async fn start_farming(farming_args: FarmingArgs) -> Result<(Farmer, Node)> {
 async fn prepare_farming() -> Result<FarmingArgs> {
     let config_args = parse_config()?;
 
-    let chain = config_args.node_config_args.chain;
     let node_name = config_args.node_config_args.name;
-    let node_directory = node_directory_getter();
-    let chain = match chain.as_str() {
+    let chain = match config_args.node_config_args.chain.as_str() {
         "gemini-2a" => chain_spec::gemini_2a().unwrap(),
         "dev" => chain_spec::dev_config().unwrap(),
         _ => unreachable!("there are no other valid chain-specs at the moment"),
     };
+    let role = match config_args.node_config_args.validator {
+        true => subspace_sdk::node::Role::Authority,
+        false => subspace_sdk::node::Role::Full,
+    };
```

Comment:
Wow, looks kinda weird. This is basically if, right?

Reply:
yeah but think about it, it is actually match. I'm not inspecting a condition, I'm matching against a value :)

Reply:
making it in an if-else looked weirder to my eyes, since it made the assumption of "this is related to the business logic of the code", which is not...